### PR TITLE
[Scale.Color] - Correctly parses color strings into rgb

### LINKED ISSUE #1527: Scale.Color produces odd results when used with named colors
Repro: http://jsfiddle.net/mj26qxxs/
Expected: Bars are red and green.
Actual: Bars are black.

Inspecting element reveals this:
screen shot 2015-01-19 at 2 41 15 pm
It appears auto-lightening code doesn't work correctly with named colors ("red", "green", etc).

### PR DESCRIPTION
**Pull request by rcchen**
Fixes #1527

### CODE REVIEW

**Comment by rcchen** [comment]
```
### Fast Compile >>typings/mocha/mocha.d.ts
Using the custom compiler : ./node_modules/typescript/bin/tsc
/Volumes/git/plottable/src/utils/utils.ts(310,7): error TS2322: Type 'Color' is not assignable to type 'HSLColor':
  Property 'h' is missing in type 'Color'.

>> 1 non-emit-preventing type warning  
>> Type errors only.
```

Also we should change darkenColor to leverage the same d3 functions; same commit or no?

**Comment by bluong** [comment]
@rcchen  Just to note that it's nice to make the pull request name to something that's a lot more readable.  I personally like the appraoch of
[{{relevantClassName}}] - {{Issue concerning that class}}

Also, I don't think the fixes tag works in pr titles?

**Comment by bluong** [comment]
@rcchen Feel free to make the darkenColor switch into a separate PR unless it's blocking this PR from going in, just in case that the darkenColor switch takes more review time that would block this fix.

**Comment by bluong** [comment]
@rcchen I think you're using commit amends here or something similar, but in any case, you should make the commit messages a bit more meaningful.  we're not at all strict about commit messages since not too much info can be derived from them, but a little information would be nice in what happened.

File: plottable.d.ts [hunk 1]
```diff
@@ -113,9 +113,9 @@ declare module Plottable {
              * @returns {[{ [key: string]: any }} coppied map.
              */
             function copyMap<T>(oldMap: {
-                [key: string]: T;
+                [x: string]: T;
```
Comment (bluong) [inline plottable.d.ts]: Seems like didn't upgrade typescript?

Comment (rcchen) [inline plottable.d.ts]:
```
rchen2-mbp:plottable rchen$ tsc -v
message TS6029: Version 1.4.1.0
```

**Comment by bluong** [comment]
LGTM aside not upgrading typescript

**Comment by crmorford** [comment]
Sorry to ask this after the dev +1...but we could do this for InterpolatedColor also? http://jsfiddle.net/mj26qxxs/2/

**Comment by bluong** [comment]
As discussed InterpolcatedColor has a colorRange API that will accomplish this, separate from this PR

**Comment by bluong** [comment]
Also as discussed, let's get the lightenFactor to a number we want

**Comment by crmorford** [comment]
Looks great! GIM

**Comment by crmorford** [comment]
I want to note that some colors are a little close, but it's a definite improvement on clarity (having 3 white squares, like before, is not good)
before: http://jsfiddle.net/u4k5t8xe/2/
after: http://jsfiddle.net/u4k5t8xe/1/

GIM

File: src/scales/colorScale.ts [hunk 2]
```diff
@@ -77,7 +77,8 @@ export module Scale {
     public scale(value: string): string {
       var color = super.scale(value);
       var index = this.domain().indexOf(value);
-      var modifyFactor = Math.floor(index / this.range().length);
+      var numLooped = Math.floor(index / this.range().length);
+      var modifyFactor = Math.log(numLooped * 1.6 + 1);
```
Comment (jtlan) [inline src/scales/colorScale.ts]: Magic 1.6? Store off as a parameter.

**Pull request merged**